### PR TITLE
Ed/hybrid builds

### PR DESCRIPTION
**Pull request by eropple**
This PR converts the library to a hybrid ESM/CJS build. ESM builds are tagged as experimental.

All tests pass, including a stub import test to make sure that the ESM default import actually works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,14 +12,21 @@
     "live streaming"
   ],
   "author": "Mux <help@mux.com> (https://mux.com/)",
-  "main": "dist",
+  "main": "./cjs/index.js",
+  "exports": "./esm/index.mjs",
```
Comment (gkatsev) [inline package.json:16]: should `exports` be more explicit?
```suggestion
  "exports": {
    "import": "./esm/index.mjs",
    "require": "./cjs/index.js"
  },
```

Comment (eropple) [reply package.json:16]: I thought that if `main` was specified, CJS ignored `exports`.

Comment (eropple) [reply package.json:16]: welp this is all terrible

File: test/unit/mux.spec.mjs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+/** @test {Mux} */
+describe('ESM: Unit::Mux', () => {
+  /** @test {Mux} */
+  describe('Mux', () => {
+    /** @test {Mux} */
+    it('exposes Mux via ESM default', () => {
+      const muxClient = new Mux('testKey', 'testSecret');
+
+      expect(muxClient).to.to.be.an.instanceof(Mux);
```
Comment (jaredsmith) [inline test/unit/mux.spec.mjs:12]: Did you really need `to.to.be` here instead of just `to.be`?

**Review by jaredsmith — APPROVED** [review]
Other than one minor comment above, everything looks good and I've encountered no issues when testing.

File: test/unit/mux.spec.mjs (hunk 2 above)
Comment (eropple) [reply test/unit/mux.spec.mjs]: Fixed

**Pull request merged**